app/src/main/java/org/systers/mentorship/view/activities/SettingsActivity.kt

```diff
@@ -20,9 +20,19 @@ class SettingsActivity : BaseActivity() {
         supportActionBar?.setDisplayShowHomeEnabled(true);
 
         tvLogout.setOnClickListener {
-            preferenceManager.clear()
-            startActivity(Intent(this, LoginActivity::class.java))
-            finishAffinity()
+            val builder = AlertDialog.Builder(this)
+            builder.setTitle(R.string.confirm_logout)
+            builder.setMessage(R.string.confirm_logout_msg)
+            builder.setPositiveButton(R.string.yes) {dialog, which ->
+                preferenceManager.clear()
```

Comment:
Please confirm if the click action is retained after configuration change (eg screen rotation).

Reply:
Yes . The app runs fine after configuration change .